homeassistant/components/sensor/transmission.py

```diff
@@ -4,89 +4,59 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/sensor.transmission/
 """
-from datetime import timedelta
 import logging
 
-import voluptuous as vol
-
-from homeassistant.components.sensor import PLATFORM_SCHEMA
-from homeassistant.const import (
-    CONF_HOST, CONF_MONITORED_VARIABLES, CONF_NAME, CONF_PASSWORD, CONF_PORT,
-    CONF_USERNAME, STATE_IDLE)
-import homeassistant.helpers.config_validation as cv
+from homeassistant.components.transmission import (
+    DATA_TRANSMISSION, SENSOR_TYPES, SCAN_INTERVAL)
+from homeassistant.const import STATE_IDLE
 from homeassistant.helpers.entity import Entity
 from homeassistant.util import Throttle
-from homeassistant.exceptions import PlatformNotReady
 
-REQUIREMENTS = ['transmissionrpc==0.11']
+DEPENDENCIES = ['transmission']
 
 _LOGGER = logging.getLogger(__name__)
 
 DEFAULT_NAME = 'Transmission'
-DEFAULT_PORT = 9091
-
-SENSOR_TYPES = {
-    'active_torrents': ['Active Torrents', None],
-    'current_status': ['Status', None],
-    'download_speed': ['Down Speed', 'MB/s'],
-    'paused_torrents': ['Paused Torrents', None],
-    'total_torrents': ['Total Torrents', None],
-    'upload_speed': ['Up Speed', 'MB/s'],
-}
-
-SCAN_INTERVAL = timedelta(minutes=2)
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_HOST): cv.string,
-    vol.Optional(CONF_MONITORED_VARIABLES, default=['torrents']):
-        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
-    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    vol.Optional(CONF_PASSWORD): cv.string,
-    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-    vol.Optional(CONF_USERNAME): cv.string,
-})
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Transmission sensors."""
-    import transmissionrpc
-    from transmissionrpc.error import TransmissionError
-
-    name = config.get(CONF_NAME)
-    host = config.get(CONF_HOST)
-    username = config.get(CONF_USERNAME)
-    password = config.get(CONF_PASSWORD)
-    port = config.get(CONF_PORT)
-
-    try:
-        transmission = transmissionrpc.Client(
-            host, port=port, user=username, password=password)
-        transmission_api = TransmissionData(transmission)
-    except TransmissionError as error:
-        if str(error).find("401: Unauthorized"):
-            _LOGGER.error("Credentials for Transmission client are not valid")
-            return
-
-        _LOGGER.warning(
-            "Unable to connect to Transmission client: %s:%s", host, port)
-        raise PlatformNotReady
+    if discovery_info is None:
+        return
+
+    component_name = DATA_TRANSMISSION
+    transmission_api = hass.data[component_name]
+    monitored_variables = discovery_info['sensors']
+    name = discovery_info['client_name']
+    sensor_types = SENSOR_TYPES
```

Comment:
We don't need to store the constant in a local variable. Use the constant directly.